R/extract.R

```diff
@@ -132,15 +132,8 @@ replace_prefix <- function(lines, prefix_pattern) {
   m <- gregexpr(prefix_pattern, lines)
   non_na <- !is.na(m)
 
-  blanks <- function(n) {
-    vapply(Map(rep.int, rep.int(" ", length(n)), n, USE.NAMES = FALSE),
-      paste, "",
-      collapse = ""
-    )
-  }
-
-  regmatches(lines[non_na], m[non_na]) <-
-    Map(blanks, lapply(regmatches(lines[non_na], m[non_na]), nchar))
+  prefix_lengths <- lapply(regmatches(lines[non_na], m[non_na]), nchar)
```

Comment:
looks OK, is it relevant to this PR?

Reply:
oh, is this fixing an existing indentation_linter violation?

Reply:
IIRC I didn't understand the original code very well and it's pretty terrible in terms of performance.
Also, not sure if indenting after `<-` should be encouraged. The old code would be more readable if `Map(` were on the preceding line.
It's such a small change that I wouldn't make a separate PR for this. LMK if I should revert.

``` r
blanks <- function(n) {
  vapply(Map(rep.int, rep.int(" ", length(n)), n, USE.NAMES = FALSE),
    paste, "",
    collapse = ""
  )
}
n <- rpois(1000, 10)
microbenchmark::microbenchmark(lapply(n, blanks), lapply(n, strrep, x = " "))
#> Unit: microseconds
#>                        expr      min        lq     mean   median        uq
#>           lapply(n, blanks) 7607.310 8014.3745 8610.423 8216.366 9332.1900
#>  lapply(n, strrep, x = " ")  578.054  607.4585  664.728  628.980  644.0755
#>        max neval
#>  11738.313   100
#>   1922.868   100
```

<sup>Created on 2022-10-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>